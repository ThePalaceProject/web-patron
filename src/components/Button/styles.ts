import { SystemStyleObject } from "@styled-system/css";
import { darken, lightness } from "@theme-ui/color";
import { ButtonSize, ButtonVariant, IconButtonVariant } from "./index";

export const sizes = {
  sm: {},
  md: {
    height: 32,
    minWidth: 32,
    fontSize: "-1",
    px: 3
  },
  lg: {
    height: 43,
    minWidth: 43,
    px: 4
  }
};

const buttonBase = {
  display: "inline-flex",
  appearance: "none",
  alignItems: "center",
  justifyContent: "center",
  textAlign: "center",
  whiteSpace: "nowrap",
  verticalAlign: "middle",
  px: 3,
  border: 0,
  borderRadius: "button",
  cursor: "pointer",
  textDecoration: "none",
  bg: "transparent"
};

export const iconButtonBase = {
  borderStyle: "none",
  cursor: "pointer",
  bg: "transparent",
  "&:focus": {
    borderWidth: "2px",
    borderStyle: "solid",
    borderColor: "ui.blue.focus",
    borderRadius: "2px"
  }
};

export const inputIconButton = {
  height: "100%",
  width: "100%"
};

export const iconButtonStyleProps = (
  variant: IconButtonVariant
): SystemStyleObject => {
  switch (variant) {
    case "input":
      return {
        ...iconButtonBase,
        ...inputIconButton
      };
    default:
      throw new Error(
        `You chose an unimplemented Icon Button Variant: ${variant}`
      );
  }
};

export const styleProps = (
  color: string,
  size: ButtonSize,
  variant: ButtonVariant
): SystemStyleObject => {
  switch (variant) {
    case "filled":
      //Prevent text color from being white if the background is set to white.
      const textColor = color === "ui.white" ? "ui.black" : "ui.white";
      return {
        // sets the text style
        variant: "text.body.regular",
        ...buttonBase,
        ...sizes[size],
        bg: color,
        color: textColor,
        fill: textColor,
        "&:focus,&:hover": {
          bg: darken(color, 0.05),
          color: textColor,
          textDecoration: "none"
        },
        "&:focus": {
          boxShadow: "focus"
        },
        "&:active": {
          bg: darken(color, 0.1)
        },
        "&:disabled": {
          bg: "ui.gray.light",
          color: "ui.gray.extraDark",
          cursor: "default"
        }
      };

    case "ghost":
      return {
        // sets the text style
        variant: "text.body.bold",
        ...buttonBase,
        ...sizes[size],
        bg: "transparent",
        color: color,
        fill: color,
        "&:focus,&:hover": {
          bg: lightness(color, 0.85),
          color: color,
          textDecoration: "none"
        },
        "&:focus": {
          boxShadow: "focus"
        },
        "&:active": {
          // bg: darken(color, 0.1)
        }
      };

    case "link":
      return {
        ...buttonBase,
        variant: "text.link.text",
        p: 0,
        color,
        fill: color,
        "&:hover": {
          color
        }
      };

    default:
      throw new Error(`You chose an unimplemented Button Variant: ${variant}`);
  }
};
