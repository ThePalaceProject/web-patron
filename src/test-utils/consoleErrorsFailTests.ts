import { jest } from "@jest/globals";
const consoleErrorSpy = jest.spyOn(global.console, "error");

// use this to make console errors fail tests. Useful when running in to issues.
export default function consoleErrorsFailTests() {
  consoleErrorSpy.mockImplementation((msg, ...opts) => {
    console.warn(msg, ...opts);
    throw new Error(msg);
  });
}
