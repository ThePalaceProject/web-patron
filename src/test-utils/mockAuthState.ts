import { AuthCredentials, OPDS1 } from "interfaces";
import Cookie from "js-cookie";
import { generateCredentials } from "utils/auth";
import { basicTokenAuthenticationUrl } from "./fixtures";

const mockCookie = Cookie as any;

export const creds: AuthCredentials = {
  token: "some-token",
  methodType: OPDS1.BasicAuthType
};

export const persistentUserCredentials = generateCredentials("1234", "pinpin");
export const firstToken = "IaMaBeArErToKeN";
export const tokenCreds1: AuthCredentials = {
  token: {
    basicToken: persistentUserCredentials,
    bearerToken: firstToken
  },
  authenticationUrl: basicTokenAuthenticationUrl,
  methodType: OPDS1.BasicTokenAuthType
};

export const newToken = "IaMaBeArErToKeN2";
export const tokenCreds2: AuthCredentials = {
  token: {
    basicToken: persistentUserCredentials,
    bearerToken: newToken
  },
  authenticationUrl: basicTokenAuthenticationUrl,
  methodType: OPDS1.BasicTokenAuthType
};

const str = JSON.stringify;

export default function mockAuthenticatedOnce(
  credentials: AuthCredentials | null = creds
) {
  mockCookie.get.mockReturnValueOnce(str(credentials));
}

export function mockAuthenticated(credentials: AuthCredentials | null = creds) {
  mockCookie.get.mockReturnValue(str(credentials));
}
