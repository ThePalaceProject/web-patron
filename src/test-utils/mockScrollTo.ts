import { jest } from "@jest/globals";
/**
 * JSdom doesn't include a stub for the scrollTo function, which we use inside of
 * Lane. Therefore we will stub it globally here
 */

Element.prototype.scrollTo = jest.fn();
window.scrollTo = jest.fn();

export default {};
