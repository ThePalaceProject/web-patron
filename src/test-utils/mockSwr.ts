import { jest } from "@jest/globals";
import { SWRResponse } from "swr";

jest.mock("swr");

export function makeSwrResponse<T>(value?: Partial<SWRResponse<T, any>>) {
  return {
    data: undefined,
    error: undefined,
    // revalidate: jest.fn(),
    isValidating: false,
    mutate: jest.fn(),
    ...value
  };
}

export type MockSwr<T> = (value?: Partial<SWRResponse<T, any>>) => void;
