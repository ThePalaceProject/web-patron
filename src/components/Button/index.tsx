/** @jsxRuntime classic */
/** @jsx jsx */
// eslint-disable-next-line @typescript-eslint/ban-ts-comment
// @ts-nocheck
import { jsx } from "theme-ui";
import { Button as BaseButton } from "@ariakit/react";
import * as React from "react";
import Link, { LinkProps } from "../Link";
import { iconButtonStyleProps, styleProps } from "./styles";
import LoadingIndicator from "components/LoadingIndicator";
import { Text } from "components/Text";
import { FontAwesomeIconProps } from "@fortawesome/react-fontawesome";

export type ButtonVariant = "filled" | "ghost" | "link";
export type IconButtonVariant = "input";
export type ButtonSize = "sm" | "md" | "lg";
type ButtonOwnProps = {
  variant?: ButtonVariant;
  size?: ButtonSize;
  color?: string;
  className?: string;
  iconLeft?: React.ComponentType;
  iconRight?: React.ComponentType;
  children: React.ReactNode;
};

/**
 * We provide three buttons that look the same, but operate differently:
 *  - Button: Renders an HTML button
 *  - NavButton: Renders a Link
 *  - AnchorButton: Renders an HTML anchor
 */
type ButtonProps = React.ComponentPropsWithoutRef<typeof BaseButton> &
  ButtonOwnProps & {
    disabled?: boolean;
    loading?: boolean;
    loadingText?: string;
  };
const Button = React.forwardRef<HTMLButtonElement, ButtonProps>(function (
  props,
  ref
) {
  const {
    variant = defaultVariant,
    color = defaultColor,
    size = defaultSize,
    loading,
    loadingText,
    disabled,
    iconLeft,
    iconRight,
    ...rest
  } = props;

  return (
    <BaseButton
      sx={styleProps(color, size, variant)}
      disabled={loading || disabled}
      ref={ref}
      aria-label={loading && loadingText ? loadingText : undefined}
      {...rest}
    >
      <ButtonContent {...props} />
    </BaseButton>
  );
});

const ButtonContent: React.FC<
  ButtonProps | NavButtonProps | AnchorButtonProps
> = ({ iconLeft: IconLeft, iconRight: IconRight, children, ...rest }) => {
  const loading = "loading" in rest ? rest.loading : false;
  const loadingText = "loadingText" in rest ? rest.loadingText : undefined;
  return (
    <>
      {!loading && IconLeft && <IconLeft sx={{ mr: 1, ml: -1 }} />}
      {!loading && children}
      {!loading && IconRight && <IconRight sx={{ mr: -1, ml: 1 }} />}
      {loading && (
        <LoadingIndicator
          color="ui.gray.extraDark"
          size={18}
          sx={{
            ml: loadingText ? -2 : undefined,
            mr: loadingText ? 2 : undefined
          }}
        />
      )}
      {loading && loadingText && <Text>{loadingText}</Text>}
    </>
  );
};
/**
 * The nav button renders a Link, which already takes an "as" prop, so
 * we need to pass that through the PolymorphicBox via some other name
 */
type NavButtonProps = LinkProps & ButtonOwnProps;
export const NavButton: JSX = React.forwardRef<
  HTMLAnchorElement,
  NavButtonProps
>(function (props, ref) {
  const {
    variant = defaultVariant,
    color = defaultColor,
    size = defaultSize,
    iconLeft,
    iconRight,
    ...rest
  } = props;
  return (
    <Link sx={styleProps(color, size, variant)} ref={ref} {...rest}>
      <ButtonContent {...props} />
    </Link>
  );
});

type AnchorButtonProps = React.ComponentPropsWithoutRef<"a"> &
  ButtonOwnProps & { newTab?: boolean };
export const AnchorButton = React.forwardRef<
  HTMLAnchorElement,
  AnchorButtonProps
>(function (props, ref) {
  const {
    color = defaultColor,
    size = defaultSize,
    variant = defaultVariant,
    newTab,
    iconLeft,
    iconRight,
    ...rest
  } = props;
  return (
    <a
      sx={styleProps(color, size, variant)}
      rel={newTab ? "noreferrer noopener" : undefined}
      target={newTab ? "__blank" : undefined}
      ref={ref}
      {...rest}
    >
      <ButtonContent {...props} />
    </a>
  );
});

interface InputIconButtonProps {
  children: React.ReactElement<FontAwesomeIconProps>;
  onClick: React.MouseEventHandler<HTMLElement>;
}
export const InputIconButton = ({
  children,
  ...props
}: InputIconButtonProps) => {
  return (
    <button sx={iconButtonStyleProps("input")} {...props}>
      {children}
    </button>
  );
};

const defaultVariant = "filled";
const defaultColor = "brand.primary";
const defaultSize = "md";

export default Button;
