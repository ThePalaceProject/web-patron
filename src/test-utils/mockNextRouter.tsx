// eslint-disable-next-line @typescript-eslint/ban-ts-comment
// @ts-nocheck
import { jest } from "@jest/globals";
import * as React from "react";
import Router from "next/router";
import { RouterContext } from "next/dist/shared/lib/router-context.shared-runtime";
import { NextRouter } from "next/router";
import { libraryData } from "test-utils/fixtures";

/**
 * Mock for the next/Router import.
 */

export const mockPush = jest.fn().mockImplementation(async () => true);
Router.push = mockPush;

export const MockNextRouterContextProvider: React.FC<{
  router?: Partial<NextRouter>;
}> = ({ router = {}, children }) => {
  const {
    isReady = true,
    basePath = "",
    route = "",
    pathname = "",
    query = {
      // add the default library slug to the default query params
      library: libraryData.slug
    },
    // default as path is the home page
    asPath = `/${libraryData.slug}`,
    push = mockPush,
    replace = jest.fn().mockImplementation(async () => true),
    reload = jest.fn().mockImplementation(() => null),
    back = jest.fn().mockImplementation(() => null),
    prefetch = jest.fn().mockImplementation(async () => undefined),
    beforePopState = jest.fn().mockImplementation(() => null),
    isFallback = false,
    isLocaleDomain = false,
    isPreview = false,
    events = {
      on: () => null,
      off: () => null,
      emit: () => null
    }
  } = router;
  return (
    <RouterContext.Provider
      value={{
        isReady,
        basePath,
        route,
        pathname,
        query,
        asPath,
        push,
        replace,
        reload,
        back,
        prefetch,
        beforePopState,
        isFallback,
        isLocaleDomain,
        isPreview,
        events
      }}
    >
      {children}
    </RouterContext.Provider>
  );
};

export default Router;
