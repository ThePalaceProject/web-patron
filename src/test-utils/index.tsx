// eslint-disable-next-line @typescript-eslint/ban-ts-comment
// @ts-nocheck
import * as React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import { LibraryData } from "../interfaces";
import "./mockScrollTo";
import * as fixtures from "./fixtures";
import serializer from "jest-emotion";
// import { Provider as ReakitProvider } from "@ariakit/react";
import { MockNextRouterContextProvider } from "./mockNextRouter";
import { NextRouter } from "next/router";
import { enableFetchMocks } from "jest-fetch-mock";
import { LibraryProvider } from "components/context/LibraryContext";
import { BreadcrumbProvider } from "components/context/BreadcrumbContext";
import { UserContext, UserState } from "components/context/UserContext";
import { ThemeUIProvider } from "theme-ui";
import makeTheme from "theme";
import mockConfig from "test-utils/mockConfig";
import track from "analytics/track";
import "react-intersection-observer/test-utils";
import "test-utils/mockToDateString";

enableFetchMocks();
expect.addSnapshotSerializer(serializer);

// standard config mock
beforeEach(() => {
  mockConfig();
});

// mock the error tracker to prevent unnecessary console logs
export const mockTrackError = jest.fn();
track.error = mockTrackError;

export { fixtures };

/**
 * mock out the window.URL.createObjectURL since it isn't
 * available on jsdom
 */
const mockCreateObjectURL = jest.fn();
(global as any)["URL"].createObjectURL = mockCreateObjectURL;

type CustomRenderOptions = Parameters<typeof render>[1] & {
  router?: Partial<NextRouter>;
  library?: Partial<LibraryData>;
  user?: Partial<UserState>;
};
/**
 * Our custom render function wraps components in mocked versions of our
 * providers
 */
const customRender = (ui: any, options?: CustomRenderOptions) => {
  const library: LibraryData = {
    ...fixtures.libraryData,
    ...options?.library
  };
  const theme = makeTheme(library.colors);

  const user: UserState = {
    ...fixtures.user,
    ...options?.user
  };

  interface AllTheProvidersProps {
    children?: React.ReactNode;
  }

  const AllTheProviders = ({ children }: AllTheProvidersProps) => {
    return (
      <MockNextRouterContextProvider router={options?.router}>
        <ThemeUIProvider theme={theme}>
          {/* <ReakitProvider> */}
          <LibraryProvider library={library}>
            <UserContext.Provider value={user}>
              <BreadcrumbProvider>{children}</BreadcrumbProvider>
            </UserContext.Provider>
          </LibraryProvider>
          {/* </ReakitProvider> */}
        </ThemeUIProvider>
      </MockNextRouterContextProvider>
    );
  };
  return {
    ...render(ui, { wrapper: AllTheProviders, ...options })
  };
};

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
