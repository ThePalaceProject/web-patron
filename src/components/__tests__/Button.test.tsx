import * as React from "react";
import { render } from "../../test-utils";
import Button, { NavButton, AnchorButton } from "../Button";

describe("variants", () => {
  test("filled (default)", () => {
    const utils = render(<Button>child</Button>);
    const button = utils.getByText("child");
    expect(button).toMatchSnapshot();
  });
  test("ghost", () => {
    const utils = render(<Button variant="ghost">child</Button>);
    const button = utils.getByText("child");
    expect(button).toMatchSnapshot();
  });
  test("link", () => {
    const utils = render(<Button variant="link">child</Button>);
    const button = utils.getByText("child");
    expect(button).toMatchSnapshot();
  });
});
test("NavButton renders correct element", () => {
  const utils = render(<NavButton href="/somewhere">child</NavButton>);
  const button = utils.getByText("child");
  expect(button).toHaveAttribute("href", "/testlib/somewhere");
  expect(button).toMatchSnapshot();
});
test("LinkButton renders correct element", () => {
  const utils = render(
    <AnchorButton href="http://some.com/external">child</AnchorButton>
  );
  const button = utils.getByText("child");
  expect(button).toHaveAttribute("href", "http://some.com/external");
  expect(button).toMatchSnapshot();
});
