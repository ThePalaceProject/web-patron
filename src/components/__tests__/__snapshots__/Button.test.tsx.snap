// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkButton renders correct element 1`] = `
.emotion-0 {
  font-family: 'Open Sans',Helvetica,Arial,sans-serif;
  font-size: 0.875rem;
  font-weight: 300;
  line-height: 1.5;
  display: inline-flex;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  text-align: center;
  white-space: nowrap;
  vertical-align: middle;
  padding-left: 16px;
  padding-right: 16px;
  border: 0;
  border-radius: 2px;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: var(--theme-ui-colors-brand-primary);
  height: 32px;
  min-width: 32px;
  color: var(--theme-ui-colors-ui-white);
  fill: var(--theme-ui-colors-ui-white);
}

.emotion-0:focus,
.emotion-0:hover {
  background-color: #2d6da3;
  color: var(--theme-ui-colors-ui-white);
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-0:focus {
  box-shadow: 0 0 0 0.2em rgba(0, 109, 255, 0.4);
}

.emotion-0:active {
  background-color: #285f8f;
}

.emotion-0:disabled {
  background-color: var(--theme-ui-colors-ui-gray-light);
  color: var(--theme-ui-colors-ui-gray-extraDark);
  cursor: default;
}

<a
  class="emotion-0"
  href="http://some.com/external"
>
  child
</a>
`;

exports[`NavButton renders correct element 1`] = `
.emotion-0 {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: var(--theme-ui-colors-ui-white);
  font-family: 'Open Sans',Helvetica,Arial,sans-serif;
  font-size: 0.875rem;
  font-weight: 300;
  line-height: 1.5;
  display: inline-flex;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  text-align: center;
  white-space: nowrap;
  vertical-align: middle;
  padding-left: 16px;
  padding-right: 16px;
  border: 0;
  border-radius: 2px;
  cursor: pointer;
  background-color: var(--theme-ui-colors-brand-primary);
  height: 32px;
  min-width: 32px;
  fill: var(--theme-ui-colors-ui-white);
}

.emotion-0:hover {
  color: inherit;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-0:focus,
.emotion-0:hover {
  background-color: #2d6da3;
  color: var(--theme-ui-colors-ui-white);
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-0:focus {
  box-shadow: 0 0 0 0.2em rgba(0, 109, 255, 0.4);
}

.emotion-0:active {
  background-color: #285f8f;
}

.emotion-0:disabled {
  background-color: var(--theme-ui-colors-ui-gray-light);
  color: var(--theme-ui-colors-ui-gray-extraDark);
  cursor: default;
}

<a
  class="emotion-0"
  href="/testlib/somewhere"
>
  child
</a>
`;

exports[`variants filled (default) 1`] = `
.emotion-0 {
  font-family: 'Open Sans',Helvetica,Arial,sans-serif;
  font-size: 0.875rem;
  font-weight: 300;
  line-height: 1.5;
  display: inline-flex;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  text-align: center;
  white-space: nowrap;
  vertical-align: middle;
  padding-left: 16px;
  padding-right: 16px;
  border: 0;
  border-radius: 2px;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: var(--theme-ui-colors-brand-primary);
  height: 32px;
  min-width: 32px;
  color: var(--theme-ui-colors-ui-white);
  fill: var(--theme-ui-colors-ui-white);
}

.emotion-0:focus,
.emotion-0:hover {
  background-color: #2d6da3;
  color: var(--theme-ui-colors-ui-white);
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-0:focus {
  box-shadow: 0 0 0 0.2em rgba(0, 109, 255, 0.4);
}

.emotion-0:active {
  background-color: #285f8f;
}

.emotion-0:disabled {
  background-color: var(--theme-ui-colors-ui-gray-light);
  color: var(--theme-ui-colors-ui-gray-extraDark);
  cursor: default;
}

<button
  class="emotion-0"
  type="button"
>
  child
</button>
`;

exports[`variants ghost 1`] = `
.emotion-0 {
  font-family: 'Open Sans',Helvetica,Arial,sans-serif;
  font-size: 0.875rem;
  font-weight: 700;
  line-height: 1.5;
  display: inline-flex;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  text-align: center;
  white-space: nowrap;
  vertical-align: middle;
  padding-left: 16px;
  padding-right: 16px;
  border: 0;
  border-radius: 2px;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: transparent;
  height: 32px;
  min-width: 32px;
  color: var(--theme-ui-colors-brand-primary);
  fill: var(--theme-ui-colors-brand-primary);
}

.emotion-0:focus,
.emotion-0:hover {
  background-color: #c3daee;
  color: var(--theme-ui-colors-brand-primary);
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-0:focus {
  box-shadow: 0 0 0 0.2em rgba(0, 109, 255, 0.4);
}

<button
  class="emotion-0"
  type="button"
>
  child
</button>
`;

exports[`variants link 1`] = `
.emotion-0 {
  display: inline-flex;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  text-align: center;
  white-space: nowrap;
  vertical-align: middle;
  padding-left: 16px;
  padding-right: 16px;
  border: 0;
  border-radius: 2px;
  cursor: pointer;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  background-color: transparent;
  font-family: 'Open Sans',Helvetica,Arial,sans-serif;
  font-size: 1rem;
  font-weight: 300;
  line-height: 1.5;
  color: var(--theme-ui-colors-brand-primary);
  padding: 0;
  fill: var(--theme-ui-colors-brand-primary);
}

.emotion-0:hover {
  color: var(--theme-ui-colors-brand-primary);
}

<button
  class="emotion-0"
  type="button"
>
  child
</button>
`;
